### PR TITLE
Allow `noServerMethods` option and otherwise throw error upon loading…

### PR DESCRIPTION
**Pull request by brettz9**
… store names which conflict with db.js methods; add test and docs

To address issue #136

### CODE REVIEW

File: tests/specs/open-db.js [hunk 1]
```diff
@@ -56,6 +56,20 @@
...
+        it('should allow avoidance of server methods', function (done) {
+            var spec = this;
+            db.open({
+                server: 'query',
+                version: 1,
+                noServerMethods: true
+            }).then(function (s) {
+                spec.server = s;
+                expect(spec.server).toBeDefined();
+                expect(spec.server.query).toEqual(jasmine.any(Function));
```
Comment (aaronpowell) [inline tests/specs/open-db.js]: I don't think the `expect` really provides much benefit. It'd be better to setup a schema and assert that the `server.<tableName>` doesn't exist.

File: tests/specs/open-db.js [hunk 2]
```diff
@@ -56,6 +56,20 @@
             });
         });
 
+        it('should allow avoidance of server methods', function (done) {
+            var spec = this;
```
Comment (aaronpowell) [inline tests/specs/open-db.js]: There's no need to capture the `this` state in the test.

File: tests/specs/open-db.js [hunk 3]
```diff
@@ -56,6 +56,20 @@
...
         });
 
+        it('should allow avoidance of server methods', function (done) {
+            var spec = this;
+            db.open({
+                server: 'query',
+                version: 1,
+                noServerMethods: true
+            }).then(function (s) {
+                spec.server = s;
```
Comment (aaronpowell) [inline tests/specs/open-db.js]: You don't need to expose the server up so there's no need to do that

File: tests/specs/open-db.js (hunk 1 above)
Comment (brettz9) [inline tests/specs/open-db.js]: Gotcha, makes sense.

**Comment by brettz9** [comment]
I've added a commit to handle the case that you suggested (and I was mistakenly checking for conflicts against the db name instead of against the object store). I did still think it was useful (via the second commit) to have a check to confirm that `query` was a function (as opposed to an object) as it ensures that `query` is still available as a server method (and not an object store), but I also added a test specifically to check that a valid store name was not present with `noServerMethods` added.

The first commit also changes things to self-close the connection since otherwise the db might remain blocked.

Let me know if you want me to revert the other changes in the first commit to reject instead of throw--I thought it might provide a clearer flow.

File: tests/specs/open-db.js [hunk 4]
```diff
@@ -56,6 +56,49 @@
...
+                    }
+                }
+            }).catch(function (err) {
+                expect(err.toString()).toContain('conflicts with db.js method');
+                done();
+            });
+        });
+
+        it('should not add stores to server using noServerMethods', function (done) {
+            var spec = this;
```
Comment (aaronpowell) [inline tests/specs/open-db.js]: No need to capture this here

File: tests/specs/open-db.js [hunk 5]
```diff
@@ -56,6 +56,49 @@
...
+                        }
+                    },
+                    query: {
+                        key: {
+                            keyPath: 'id'
+                        }
+                    }
+                }
+            }).then(function (s) {
+                spec.server = s;
```
Comment (aaronpowell) [inline tests/specs/open-db.js]: don't need this here, just assert off `s`

File: tests/specs/open-db.js (hunk 4 above)
Comment (brettz9) [inline tests/specs/open-db.js]: Why not confirm that an error is thrown?

File: tests/specs/open-db.js (hunk 5 above)
Comment (brettz9) [inline tests/specs/open-db.js]: I set spec.server so that the `afterEach` had the server to close...Otherwise, the next test got blocked...

**Comment by brettz9** [comment]
Are you ok with my further modifications here?

File: tests/specs/open-db.js (hunk 5 above)
Comment (aaronpowell) [inline tests/specs/open-db.js]: Ah good point, forgot that that needs to be done!

**Pull request merged**